engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.br;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraints.Mod11Check;
+import org.hibernate.validator.constraints.br.CNPJ;
+import org.hibernate.validator.internal.constraintvalidators.hv.Mod11CheckValidator;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationDescriptor;
+import org.hibernate.validator.internal.util.annotationfactory.AnnotationFactory;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class CNPJValidator implements ConstraintValidator<CNPJ, CharSequence> {
+	private final Mod11CheckValidator firstMod11CheckWithSeparator = new Mod11CheckValidator();
+	private final Mod11CheckValidator secondMod11CheckWithSeparator = new Mod11CheckValidator();
+
+	private final Mod11CheckValidator firstMod11CheckWithoutSeparator = new Mod11CheckValidator();
+	private final Mod11CheckValidator secondMod11CheckWithoutSeparator = new Mod11CheckValidator();
+
+	@Override
+	public void initialize(CNPJ constraintAnnotation) {
+		firstMod11CheckWithSeparator.initialize( createMod11ProxyAnnotation( 9, 14, 16, true ) );
```

Comment:
> How about adding an initialize() method to Mod11CheckValidator which takes the actual argument types instead of the annotation. That'd be a bit more straight-forward than creating the annotation proxy just for retrieving the values from it.

Interesting idea. I wanted to stay with the constraint validator API and not add new methods, but since this the validator instance is internal anyways, it would not matter. Also this felt like constraint composition, just that I could not to it completely via configuration. I'll look into it.